### PR TITLE
Replace Cloudfront with Vercel rewrites for demo.saleor.io

### PR DESCRIPTION
**Pull request by witoszekdev**
This PR adds Vercel rewrites instead of Cloudfront for making requests to Saleor Cloud (e.g. `/graphql` route)

### CODE REVIEW

File: apps/storefront/vercel.json [hunk 1]
```diff
@@ -0,0 +1,43 @@
+{
+  "$schema": "https://openapi.vercel.sh/vercel.json",
+  "redirects": [
+    {
+      "source": "/grpahql",
```
Comment (mmiszy) [inline apps/storefront/vercel.json:5]: Is this typo on purpose?

Comment (witoszekdev) [reply apps/storefront/vercel.json]: no 🤦, fixed

**Comment by witoszekdev** [comment]
Due to issues with infinite redirects (`/graphql/` -> `/graphql` -> `/graphql/` -> etc.) caused by default Vercel confgiuration, we'll keep on using our custom CloudFront for the demo.

**Comment by witoszekdev** [comment]
let's try this again - with Next.js config instead

File: apps/storefront/vercel.json [hunk 2]
```diff
@@ -0,0 +1,39 @@
+{
```
Comment (mmiszy) [inline apps/storefront/vercel.json:1]: Does it work fine with the async rewrites() { we have in next.config.js?
Can we use one of them and not both?

Comment (witoszekdev) [reply apps/storefront/vercel.json:1]: I think we can use both, as `https://dev-demo.saleor.io/checkout/` works fine. But maybe we should actually just use the Next.js redirects. I'll give it a try :)

Comment (witoszekdev) [reply apps/storefront/vercel.json]: It works fine 👍

**Review by mmiszy — APPROVED** [review]

**Pull request merged**